src/components/Activity/ActivityList.js

```diff
@@ -24,68 +24,96 @@ const ActivityList = ({ apps, activities, keys, clearActivity }) => {
-          trail={100}
-          from={{ opacity: 0, height: 0, transform: 'translate3d(-20px,0,0)' }}
-          enter={item => async next => {
-            await next({ height: 'auto' })
-            await next({ opacity: 1, transform: 'translate3d(0px,0,0)' }, true)
+  const transitionKeys = activity => activity.transactionHash
 
-            const activityKey = keys(item)
-            setActivitiesReady({ ...activitiesReady, [activityKey]: true })
-          }}
-          leave={[{ opacity: 1 }, { height: 0 }]}
-          config={[{ ...springs.swift, precision: 1 }, springs.swift]}
-        >
-          {(activity, state) => props => {
-            return (
-              <animated.div style={{ ...props, overflow: 'hidden' }}>
-                <ActivityItem
-                  activity={activity}
-                  onClose={() => {
-                    clearActivity(activity.transactionHash)
-                  }}
-                />
-              </animated.div>
-            )
-          }}
-        </Transition>
-      ) : (
-        <div
-          css={`
-            display: flex;
-            flex-direction: column;
-            align-items: center;
-            width: 100%;
-            padding: 64px 32px 24px;
-          `}
-        >
-          <IconEmptyState />
-          <p
-            css={`
-              margin-top: 10px;
-              font-size: 14px;
-              color: ${theme.textSecondary};
-            `}
-          >
-            No activity yet.
-          </p>
-        </div>
-      )}
+  return (
+    <div css="position: relative">
+      <Transition
+        native
+        items={activityItems.length === 0}
+        from={{ opacity: 0 }}
+        enter={{ opacity: 1 }}
+        leave={{ opacity: 0 }}
+        config={springs.smooth}
+      >
+        {show =>
+          show &&
+          (transitionStyles => (
+            <div
+              css={`
+                position: absolute;
+                top: 0;
+                width: 100%;
+                display: flex;
+                flex-direction: column;
+                align-items: center;
+                padding: 64px 32px 24px;
+              `}
+              style={transitionStyles}
+            >
+              <IconEmptyState />
+              <p
+                css={`
+                  margin-top: 10px;
+                  font-size: 14px;
+                  color: ${theme.textSecondary};
+                `}
+              >
+                No activity.
+              </p>
+            </div>
+          ))
+        }
+      </Transition>
+      <Transition
+        native
+        items={activityItems}
+        keys={transitionKeys}
+        trail={100}
+        initial={{
+          opacity: 1,
+          height: 'auto',
+          transform: 'translate3d(0px, 0, 0)',
+        }}
+        from={{
+          opacity: 0,
+          height: 0,
+          transform: 'translate3d(-20px, 0, 0)',
+        }}
+        enter={item => async next => {
```

Comment:
Does this fall into the category of declaring functions every-time this is rendered?

Reply:
It does yes, but I don’t think it’s an issue here?

Reply:
Hmmm, not sure now, with hooks I don't think declaring functions over and over is heavy (because classes were heavier) and we have `useCallback`. I asked to see if you had any insights here.

Reply:
No I don’t really see an issue apart from declaring a function, but yes it’s pretty cheap :+1: